redbeams/src/main/java/com/sequenceiq/redbeams/flow/redbeams/stop/handler/StopDatabaseServerHandler.java

```diff
@@ -0,0 +1,68 @@
+package com.sequenceiq.redbeams.flow.redbeams.stop.handler;
+
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.scheduler.SyncPollingScheduler;
+import com.sequenceiq.cloudbreak.cloud.task.PollTaskFactory;
+import com.sequenceiq.cloudbreak.cloud.task.ResourcesStatePollerResult;
+import com.sequenceiq.flow.event.EventSelectorUtil;
+import com.sequenceiq.flow.reactor.api.handler.EventHandler;
+import com.sequenceiq.redbeams.flow.redbeams.common.RedbeamsEvent;
+import com.sequenceiq.redbeams.flow.redbeams.stop.event.StopDatabaseServerFailed;
+import com.sequenceiq.redbeams.flow.redbeams.stop.event.StopDatabaseServerRequest;
+import com.sequenceiq.redbeams.flow.redbeams.stop.event.StopDatabaseServerSuccess;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import reactor.bus.Event;
+import reactor.bus.EventBus;
+
+@Component
+public class StopDatabaseServerHandler implements EventHandler<StopDatabaseServerRequest> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StopDatabaseServerHandler.class);
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private PollTaskFactory statusCheckFactory;
+
+    @Inject
+    private SyncPollingScheduler<ResourcesStatePollerResult> syncPollingScheduler;
+
+    @Inject
+    private EventBus eventBus;
+
+    @Override
+    public String selector() {
+        return EventSelectorUtil.selector(StopDatabaseServerRequest.class);
+    }
+
+    @Override
+    public void accept(Event<StopDatabaseServerRequest> event) {
+        LOGGER.debug("Received event: {}", event);
+        StopDatabaseServerRequest request = event.getData();
+        CloudContext cloudContext = request.getCloudContext();
+        try {
+            CloudConnector<Object> connector = cloudPlatformConnectors.get(cloudContext.getPlatformVariant());
+            AuthenticatedContext ac = connector.authentication().authenticate(cloudContext, request.getCloudCredential());
+
+            connector.resources().stopDatabaseServer(ac, request.getDbInstanceIdentifier());
```

Comment:
I think we should check first that do we really need to start the initiate stop action or not.
When one of the CB node dies and trying to restart the flow we dont want to initiate the stop event twice

Reply:
Its already done in https://github.com/hortonworks/cloudbreak/blob/fbe8818ef58b368634efdc3e7e16b282435014b9/redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/RedbeamsStopService.java#L43

Should I implement it on AWS level as well?

Reply:
not on aws level but in the abstract flow step

Reply:
Done.